app/helpers/Emails/NotificationsHelper/Comments/SolutionCommentsEmailsSender.php

```diff
@@ -0,0 +1,142 @@
+    $this->emailHelper = $emailHelper;
+    $this->sender = Arrays::get($params, ["emails", "from"], "noreply@recodex.cz");
+    $this->assignmentSolutionCommentPrefix = Arrays::get($params, ["emails", "assignmentSolutionCommentPrefix"], "Assignment Solution Comment - ");
+    $this->referenceSolutionCommentPrefix = Arrays::get($params, ["emails", "referenceSolutionCommentPrefix"], "Reference Solution Comment - ");
+  }
+
+  /**
+   * Internal solution send comment method.
+   * @param AssignmentSolution|ReferenceExerciseSolution $solution
+   * @param Comment $comment
+   * @return bool
+   */
+  private function sendSolutionComment($solution, Comment $comment): bool {
+    if ($comment->isPrivate()) {
+      // comment was private, therefore do not send email to others
+      return true;
+    }
+
+    $baseSolution = $solution->getSolution();
+    $subject = $this->assignmentSolutionCommentPrefix . $baseSolution->getAuthor()->getName();
+    $recipients = [$baseSolution->getAuthor()->getEmail()];
+    foreach ($comment->getThread()->findAllPublic() as $pComment) {
+      $user = $pComment->getUser();
+      if (!$user->getSettings()->getSolutionCommentsEmails()) {
+        continue;
+      }
+      $recipients[] = $user->getEmail();
+    }
+
+    // make user emails unique, so the emails won't be multiplied
+    $recipients = array_unique($recipients);
+
+    // filter out the author of the comment, it is pointless to send email to that user
+    $recipients = array_filter($recipients, function (string $email) use ($comment) {
+      return $email !== $comment->getUser()->getEmail();
+    });
+
+    if (count($recipients) === 0) {
+      return true;
+    }
+
+    if ($solution instanceof AssignmentSolution) {
+      $body = $this->createAssignmentSolutionCommentBody($solution, $comment);
+    } else {
+      $body = $this->createReferenceSolutionCommentBody($solution, $comment);
+    }
+
+    // Send the mail
+    return $this->emailHelper->send(
+      $this->sender,
+      [],
+      $subject,
+      $body,
+      $recipients
+    );
+  }
+
+  /**
+   * Comment was added to the assignment solution.
+   * @param AssignmentSolution $solution
+   * @param Comment $comment
+   * @return boolean
+   */
+  public function assignmentSolutionComment(AssignmentSolution $solution, Comment $comment): bool {
+    return $this->sendSolutionComment($solution, $comment);
+  }
+
+  /**
+   * Prepare and format body of the assignment solution comment.
+   * @param AssignmentSolution $solution
+   * @param Comment $comment
+   * @return string Formatted mail body to be sent
+   */
+  private function createAssignmentSolutionCommentBody(AssignmentSolution $solution, Comment $comment): string {
+    // render the HTML to string using Latte engine
+    $latte = new Latte\Engine();
+    return $latte->renderToString(__DIR__ . "/assignmentSolutionComment.latte", [
+      "assignment" => $solution->getAssignment()->getLocalizedTexts()->first()->getName(), // TODO
```

Comment:
Well, the template is in English, I would suggest to get English name if available, then resort to the first localization.